src/lib/provable/int.ts

```diff
@@ -401,56 +408,55 @@ class UInt64 extends CircuitValue {
   lessThanOrEqual(y: UInt64) {
     if (this.value.isConstant() && y.value.isConstant()) {
       return Bool(this.value.toBigInt() <= y.value.toBigInt());
-    } else {
-      let xMinusY = this.value.sub(y.value).seal();
-      let yMinusX = xMinusY.neg();
-
-      let xMinusYFits = RangeCheck.isDefinitelyInRangeN(
-        UInt64.NUM_BITS,
-        xMinusY
-      );
-
-      let yMinusXFits = RangeCheck.isDefinitelyInRangeN(
-        UInt64.NUM_BITS,
-        yMinusX
-      );
-
-      xMinusYFits.or(yMinusXFits).assertEquals(true);
-      // x <= y if y - x fits in 64 bits
-      return yMinusXFits;
     }
+    return lessThanOrEqualGeneric(this.value, y.value, 1n << 64n, (v) =>
+      RangeCheck.rangeCheckN(UInt64.NUM_BITS, v)
+    );
```

Comment:
this used to be less efficient because it called `rangeCheckHelper()` twice. the new version only calls it once